src/twisted/test/test_defer.py

```diff
@@ -1415,19 +1415,19 @@ def ic(d):
         newFailure = self.failureResultOf(d)
         tb = traceback.extract_tb(newFailure.getTracebackObject())
 
-        self.assertEqual(len(tb), 2)
+        if _PY3:
+            self.assertEqual(len(tb), 3)
+            self.assertIn('test_defer', tb[2][0])
+            self.assertEqual('getDivisionFailure', tb[2][2])
+            self.assertEqual('1/0', tb[2][3])
```

Comment:
Out of curiosity, did you consider using the filename, line, and name attributes of the FrameSummary object?  As the objects are new in 3.5 that wouldn't work if Twisted supported versions pre 3.5, but I'm drawing a blank on whether that's true (doesn't look like anything pre 3.5 in python3 is in the build list).

Not a concern, it just might be more readable than the indicies into the FrameSummary object.